code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -86,6 +86,58 @@ where
 	Ok(ao)
 }
 
+/// Compute the amount of the output token given the amount of the input token.
+///
+/// If `Ok`, returns a tuple containing `(a_out, fee)`.
+/// To get `a_out` without accounting for the fee, set `f = 0`.
+///
+/// **NOTE:** Weights must already be normalized.
+///
+/// # Parameters
+/// * `w_i` - Weight of the input token
+/// * `w_o` - Weight of the output token
+/// * `b_i` - Balance of the input token
+/// * `b_o` - Balance of the output token
+/// * `a_sent` - Amount of the input token sent by the user
+/// * `f` - Total swap fee
+///
+/// From https://github.com/ComposableFi/composable/blob/main/rfcs/0008-pablo-lbp-cpp-restructure.md#41-fee-math-updates,
+/// equation (2)
+pub fn compute_out_given_in_new<T: PerThing>(
```

Comment:
Should round down overall for OutGivenIn
https://github.com/balancer-labs/balancer-v2-monorepo/blob/592b0534f782b9bfb668990f7821e041a9b775d4/pkg/pool-weighted/contracts/WeightedMath.sol#L95

Reply:
7b5b4cabd92cfac17dc1c0b66b5e4cd85e92dce7